### PR TITLE
Key vault IP Restriction for Cosmos apps

### PR DESCRIPTION
**Pull request by reneenoble**
Restrict access of IPs for Key Vault in Cosmos templates

As per https://github.com/Azure-Samples/Azure-Python-Standardization-Template-Generator/security/code-scanning/21

### CODE REVIEW

File: {{cookiecutter.__src_folder_name}}/infra/main.bicep [hunk 1]
```diff
@@ -47,6 +51,11 @@ module keyVault './core/security/keyvault.bicep' = {
     location: location
     tags: tags
     principalId: principalId
+    allowPublicNetworkAccess: false
+    networkAcls: {
+      allowAzureServicesAccess: allowAzureServicesAccess
+      defaultAction: defaultAction
+    }
```
Comment (reneenoble) [inline {{cookiecutter.__src_folder_name}}/infra/main.bicep:58]: I don't know if it is needed to add this section here provide the params or if we just want it to fall back to the default. 
In terms of patterns is it better to put it in the main.bicep file that there are these options so its up front, or to stash them away?

Comment (tonybaloney) [reply {{cookiecutter.__src_folder_name}}/infra/main.bicep:58]: It's not required, remove it from this file

Comment (reneenoble) [reply {{cookiecutter.__src_folder_name}}/infra/main.bicep]: OK cool have removed

File: {{cookiecutter.__src_folder_name}}/infra/core/security/keyvault.bicep [hunk 2]
```diff
@@ -5,13 +5,27 @@
...
+}
+
 resource keyVault 'Microsoft.KeyVault/vaults@2022-07-01' = {
   name: name
   location: location
   tags: tags
   properties: {
     tenantId: subscription().tenantId
     sku: { family: 'A', name: 'standard' }
+    networkAcls: networkAcls
```
[the remark this answers is not in the document]
Comment (reneenoble) [reply {{cookiecutter.__src_folder_name}}/infra/core/security/keyvault.bicep:28]: Just want to check that this is taken care of by the bypass param? @tonybaloney 

```  bypass: allowAzureServicesAccess ? 'AzureServices' : 'None'
  defaultAction: allowPublicNetworkAccess ? 'Allow' : 'Deny'
  ipRules: []
  virtualNetworkRules: []```

File: {{cookiecutter.__src_folder_name}}/infra/core/security/keyvault.bicep [hunk 3]
```diff
@@ -5,13 +5,27 @@ param tags object = {}
 
 param principalId string = ''
 
+// Allow public network access to Key Vault
+param allowPublicNetworkAccess bool = true
```
Comment (tonybaloney) [inline {{cookiecutter.__src_folder_name}}/infra/core/security/keyvault.bicep:9]: We're going to default this to `false` now.

File: {{cookiecutter.__src_folder_name}}/infra/main.bicep [hunk 4]
```diff
@@ -32,6 +32,10 @@ var resourceToken = toLower(uniqueString(subscription().id, name, location))
 var prefix = '${name}-${resourceToken}'
 var tags = { 'azd-env-name': name }
 
+var allowAzureServicesAccess = true
```
Comment (tonybaloney) [inline {{cookiecutter.__src_folder_name}}/infra/main.bicep:35]: The changes to this file aren't required

**Review by tonybaloney — APPROVED** [review]

**Pull request merged**